File: csharp/Domain/Models/Team.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Models
{
    public class Team : Identifiable<long>
    {
        public String TeamName { get; set; }

        public Team(string teamName)
        {
            TeamName = teamName;
        }

        public override bool Equals(object? obj)
        {
            return obj is Team team &&
                   base.Equals(obj) &&
                   Id == team.Id &&
                   TeamName == team.TeamName;
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(base.GetHashCode(), Id, TeamName);
        }
    }
}

File: java/Domain/src/main/java/com/rallies/domain/models/Team.java
package com.rallies.domain.models;

import java.util.Objects;

public class Team extends Identifiable<Long> {
    private String teamName;

    public Team(String teamName) {
        this.teamName = teamName;
    }

    public Team(Team other) {
        this.teamName = other.teamName;
    }

    public String getTeamName() {
        return teamName;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Team)) return false;
        if (!super.equals(o)) return false;
        Team team = (Team) o;
        return Objects.equals(teamName, team.teamName);
    }

    @Override
    public String toString() {
        return "Team{" +
                "teamName='" + teamName + '\'' +
                '}';
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), teamName);
    }
}
